### DISCUSSION TITLE
Title: How to set SOC-based thresholds in PyBaMM Experiments?

### DISCUSSION THREAD

**Discussion by ErfanSamadi1998** [pybamm-team/PyBaMM#4493] [Q&A]
Hi PyBaMM team,

I've been working with pybamm.Experiment and noticed that, while it's possible to control the experiment with time, current, or voltage thresholds, there doesn't seem to be a direct way to set SOC-based thresholds for starting or stopping charge/discharge.

For example, something like:

"Discharge at 3C until 0.5 SOC"
"Charge at 1C until 0.9 SOC"
I know I can set the initial SOC for a simulation using sim.solve(initial_soc=1), but for running experiments, the SOC-based stopping conditions don't seem to be available.

Am I missing something here? Is there a way to set SOC as a stopping condition in pybamm.Experiment or a recommended workaround for achieving this?

Thank you for your help!

**Comment by pkalbhor** [comment]
There is no official implementation of SoC in PyBaMM AFAIK, see this comment by @valentinsulzer  for possible implemntation https://github.com/pybamm-team/PyBaMM/issues/2553#issuecomment-1351551806

Other logical difficulty in implementation of SoC based limits for PyBaMM experiment is that it will hit cutoff voltage before SoC limit(unless c-rate is very low or SoC limit is high). 

> Is there a way to set SOC as a stopping condition in pybamm.Experiment?

No

**Reply by ErfanSamadi1998** [reply]
Thank you for the clarification! I understand now. Appreciate your help!

**Comment by rtimms** [comment]
You can define custom termination conditions, see https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/examples/notebooks/simulations_and_experiments/custom-experiments.ipynb. So you just need to define a new variable for however you SOC.

**Comment by ErfanSamadi1998** [comment]
Thank you so much for pointing me to the custom termination conditions and the example notebook! I'll explore it and try defining a new variable for SOC as suggested. I really appreciate your guidance!
